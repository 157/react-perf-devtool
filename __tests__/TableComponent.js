var React = require('react')
var ReactTestRenderer = require('react-test-renderer')

var Table = require('../src/extension/components/Table')

var rawMeasures = require('../samples/measures')

var getReactPerformanceData = require('../src/shared/parse')
var generateDataFromMeasures = require('../src/shared/generate')

var aggregatedMeasures = generateDataFromMeasures(
  getReactPerformanceData(rawMeasures)
)

describe('Table component', () => {
  it('should output the calculated results in a table', () => {
    var tree = ReactTestRenderer.create(
      <Table measures={aggregatedMeasures} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
