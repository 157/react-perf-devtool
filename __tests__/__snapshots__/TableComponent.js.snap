// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should output the calculated results in a table 1`] = `
<table
  id="measures"
>
  <tr>
    <th>
      Components
    </th>
    <th>
      Total time (ms)
    </th>
    <th>
      Count
    </th>
    <th>
      Total time (%)
    </th>
    <th>
      Mount (ms)
    </th>
    <th>
      Update (ms)
    </th>
    <th>
      Render (ms)
    </th>
    <th>
      Unmount (ms)
    </th>
    <th>
      componentWillMount (ms)
    </th>
    <th>
      componentDidMount (ms)
    </th>
    <th>
      componentWillReceiveProps (ms)
    </th>
    <th>
      shouldComponentUpdate (ms)
    </th>
    <th>
      componentWillUpdate (ms)
    </th>
    <th>
      componentDidUpdate (ms)
    </th>
    <th>
      componentWillUnmount (ms)
    </th>
  </tr>
  <tr>
    <td>
      App
    </td>
    <td>
      40.82
    </td>
    <td>
      1
    </td>
    <td>
      35%
    </td>
    <td>
      36.88
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      3.02
    </td>
    <td>
      0.92
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
  </tr>
  <tr>
    <td>
      Main
    </td>
    <td>
      33.04
    </td>
    <td>
      1
    </td>
    <td>
      29%
    </td>
    <td>
      31.35
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0.88
    </td>
    <td>
      0.81
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
  </tr>
  <tr>
    <td>
      Introduction
    </td>
    <td>
      16.27
    </td>
    <td>
      1
    </td>
    <td>
      14%
    </td>
    <td>
      11.33
    </td>
    <td>
      2.83
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0.84
    </td>
    <td>
      0.46
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0.81
    </td>
    <td>
      0
    </td>
  </tr>
  <tr>
    <td>
      Projects
    </td>
    <td>
      14.4
    </td>
    <td>
      1
    </td>
    <td>
      13%
    </td>
    <td>
      9.44
    </td>
    <td>
      3.03
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      1.85
    </td>
    <td>
      0.08
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
  </tr>
  <tr>
    <td>
      Posts
    </td>
    <td>
      5.99
    </td>
    <td>
      1
    </td>
    <td>
      5%
    </td>
    <td>
      4.05
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      1.11
    </td>
    <td>
      0.83
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
  </tr>
  <tr>
    <td>
      Message
    </td>
    <td>
      4.63
    </td>
    <td>
      1
    </td>
    <td>
      4%
    </td>
    <td>
      3.41
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0.67
    </td>
    <td>
      0.55
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
    <td>
      0
    </td>
  </tr>
</table>
`;
